libbeat/common/transport/httpcommon/proxy.go

```diff
@@ -47,6 +50,46 @@ type HTTPClientProxySettings struct {
 	Disable bool `config:"proxy_disable" yaml:"proxy_disable,omitempty"`
 }
 
+// MarshalYAML serializes URI as a string.
+func (p ProxyURI) MarshalYAML() (interface{}, error) {
+	u := url.URL(p)
+	return u.String(), nil
+}
+
+// URI returns conventional url.URL structure.
+func (p ProxyURI) URI() *url.URL {
+	uri := url.URL(p)
+	return &uri
+}
```

Comment:
nit; this is just a shallow copy. How about `func (p *ProxyURI) URI() *url.URL { return (*url.URL)(p) }`